pandas/plotting/_matplotlib/timeseries.py

```diff
@@ -251,7 +251,7 @@ def _maybe_convert_index(ax, data):
         freq = frequencies.get_period_alias(freq)
 
         if isinstance(data.index, ABCDatetimeIndex):
-            data = data.to_period(freq=freq)
+            data = data.tz_localize(None).to_period(freq=freq)
```

Comment:
To preserve the existing behaviour, this might need to be `tz_convert` instead of `tz_localize` ? (the first will give the underlying UTC data, the other converts to naive local time)

Reply:
I would maybe check with a small actual example and see what plotting it gives (on 0.25.3 and on this PR).

Because intuitively, converting to local time zone sounds more useful, though. But not fully sure what the current behaviour is

Reply:
Here's the output of
```
df = pd.DataFrame(np.random.randn(100, 3), index=pd.date_range("2012", freq='H', periods=100, tz='UTC'), columns=['a', 'b', 'c']) 
df.head().plot() 
```
on 0.25.3:
![image](https://user-images.githubusercontent.com/33491632/73071117-e922f980-3ea9-11ea-8201-dd8e012c9915.png)

On this branch:
![image](https://user-images.githubusercontent.com/33491632/73071170-0eb00300-3eaa-11ea-8d6d-d6a333e0f4c5.png)

Reply:
@MarcoGorelli can you try with something else as UTC? (as UTC is the one timezone where UTC and local is the same :-))

Reply:
@jorisvandenbossche sure :)

```
df = pd.DataFrame(
    np.random.randn(100, 3),
    index=pd.date_range(
        "2012", freq="H", periods=100, tz=pytz.timezone("Africa/Gaborone")
    ),
    columns=["a", "b", "c"],
)
df.head().plot()
```

0.25.3
![image](https://user-images.githubusercontent.com/33491632/73119824-3e6d1280-3f5f-11ea-849a-5f1aec1a4a5b.png)


This branch:
![image](https://user-images.githubusercontent.com/33491632/73119807-17164580-3f5f-11ea-9a9a-8816467a37e3.png)